packages/now-cli/src/util/env/get-env-records.ts

```diff
@@ -1,18 +1,42 @@
 import { Output } from '../output';
 import Client from '../client';
-import { ProjectEnvVariable, ProjectEnvTarget } from '../../types';
+import {
+  ProjectEnvVariable,
+  ProjectEnvTarget,
+  PaginationOptions,
+} from '../../types';
+
+export type APIV5Response = {
+  envs: ProjectEnvVariable[];
+  pagination: PaginationOptions;
+};
+
+export type APIV4Response = ProjectEnvVariable[];
 
 export default async function getEnvVariables(
   output: Output,
   client: Client,
   projectId: string,
-  target?: ProjectEnvTarget
-): Promise<ProjectEnvVariable[]> {
+  target?: ProjectEnvTarget,
+  next?: number,
+  apiVersion: number = 4
```

Comment:
Let's make this `apiVersion` argument determine the return type of the function so the other calling functions don't need to cast the type.

Reply:
Example TS solution: https://www.typescriptlang.org/play/index.html#code/PTAEEMBtNAHB7AzoglgIxZFAXFBTRAKGwE9Y9QBBWFANTwCdV4A7UAXlABZQAfUAKwBuQoRChUAW1hYAZvgAmoPAA9w0yAVABXVCwDmE7AxQHEEUGnjxN4FudnwGYsAryTWiY+Fys42hgREAmIyCkoABQBJWi4AJQIEewpOLxMDAG0AXRFCU2xGWXAAY3Do2gEExCTg0ABvQlA4cH1THxRWAC5La1sWESa8FgA3RG60031skQBfUXFwBSV4YcZIeEXzbAALH1BJcFhQVaYOtmx4UAY8bADzsMJZbRZi3zZ9G4BREdpwE3A0JpEAAKE7MFjdLgASm6kRi8USnjwIieLzeoA+2G+w1+-0BBFBjHB3QEMKo5UqiOSuXEKA07iG2HarEez1eZwxXx+fxQAKBAB5aMoVAUWApzNQ6ESzgA+QmnLqgWhQ+qNUAoWSgeXgjjsTik1VNJrFTzYK4EbSQbCwilVGoperNVosZkQ0DGbR4AA0yhGY1AGQA5CwUKVA1lQDMBkbrrcGGxrohLdhozNlJBahqtWCOXrONDDUaTfYzYnkzb4XakRwA8HQ3hw9GmrG7uak1bU+nag0je7tgx4AB3UAsPDDz4MAcMYGBgCqLAA1iwh2wc4rA6AANTHaWsKGpwhzFzC9QyCi6Fp4QjFrxt5M8TiY7G43n4kHQkQ30sWq0CGtP7k8SBYFSVyLNgTLK0uAAOk0Awdl1TgAEYVR7UBxEHJwF0QABCQ9RHAyDsAEaDYBaNo3lQtUFgzS5MIYbC8JmIA